tests/dat/actions/wcbin.js

```diff
@@ -5,25 +5,26 @@
 function main(params) {
     var str = params.payload;
     console.log("The payload is '" + str + "'");
-    whisk.invoke({
-        name : 'wc',
-        parameters : {
-            payload : str
-        },
-        blocking : true,
-        next : function(error, activation) {
+
+    var promise = new Promise(function(resolve, reject) {
+        whisk.invoke({
```

Comment:
@jasonpet What do you think about this instead? I _believe_ it is functionally the same (minus one `console.log()` in the error condition) and does not require you to explicitly instantiate a promise:

```javascript
function main(params) {
    var str = params.payload;
    console.log("The payload is '" + str + "'");

    return whisk.invoke({
            name: 'wc',
            parameters: { payload: str },
            blocking: true})
      .then(function (activation) {
            console.log('activation:', activation);
            var wordsInDecimal = activation.result.count;
            var wordsInBinary = wordsInDecimal.toString(2) + ' (base 2)';

            return { binaryCount: wordsInBinary };
        });
    });
}
```